sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -180,11 +189,13 @@ object ScalaReflection extends ScalaReflection {
    * @param tpe The `Type` of deserialized object.
    * @param path The expression which can be used to extract serialized value.
    * @param walkedTypePath The paths from top to bottom to access current field when deserializing.
+   * @param instantiateValueClass If `true`, create an instance for Scala value class
```

Comment:
it will be good to explain when we need to instantiate value class and why

Reply:
ok, I updated the comment